### PR TITLE
Avoid endless loop at magic auth

### PR DESCRIPTION
**Pull request by annando**
The last days I had massive problems with one of my servers. Under certain circumstances it came to an endless loop at the magic auth. According to my logs this is solved now. And hopefully magic auth will still work.

### CODE REVIEW

File: src/Model/Profile.php [hunk 1]
```diff
@@ -1010,56 +1010,59 @@ public static function zrlInit(App $a)
...
-				// Is it a DDoS attempt?
-				// The check fetches the cached value from gprobe to reduce the load for this system
-				$urlparts = parse_url($my_url);
-
-				$result = Cache::get('gprobe:' . $urlparts['host']);
-				if ((!is_null($result)) && (in_array($result['network'], [Protocol::FEED, Protocol::PHANTOM]))) {
-					logger('DDoS attempt detected for ' . $urlparts['host'] . ' by ' . $_SERVER['REMOTE_ADDR'] . '. server data: ' . print_r($_SERVER, true), LOGGER_DEBUG);
-					return;
-				}
+		if (empty($my_url) || local_user()) {
```
Comment (rabuzarus) [inline src/Model/Profile.php]: Why shouldn't the `local_user()` authenticate if he/she visit a local guest profile?

**Comment by rabuzarus** [comment]
> Under certain circumstances it came to an endless loop at the magic auth.

Can you describe the circumstances which result in an endless loop?

**Comment by annando** [comment]
The magic auth request resulted in some magic auth request on the other side that created some magic auth request. I guess the reason had been some check I added for the last version that checked for the existence of magic auth on the other side.

**Comment by MrPetovan** [comment]
Could explain the high load on everyone's server on `develop`.

**Comment by annando** [comment]
But I still don't understand why it is triggered now. Possibly there had been some other change that initiated it.

**Pull request merged**